### PR TITLE
jbide-24782: NullPointerException in ResourceUtils.getBuildConfigFor

### PR DESCRIPTION
**Pull request by bdshadow**
# Pull Request Checklist
## General
* Is this a blocking issue or new feature? If yes, QE needs to +1 this PR

## Code
* Are method-/class-/variable-names meaningful?
* Are methods concise, not too long?
* Are catch blocks catching precise Exceptions only (no catch all)?

## Testing
* Are there unit-tests?
* Are there integration tests (or at least a jira to tackle these)?
* Is the non-happy path working, too?
* Are other parts that use the same component still working fine?

## Function
* Does it work?

### CODE REVIEW

File: plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/server/ServerSettingsWizardPage.java [hunk 1]
```diff
@@ -497,9 +495,9 @@ private SelectionListener onImportProject(ServerSettingsWizardPageModel model, f
             @Override
             public void widgetSelected(SelectionEvent e) {
                 Map<com.openshift.restclient.model.IProject, Collection<IBuildConfig>> projectsAndBuildConfigs = new HashMap<>();
-                List<IBuildConfig> buildConfigs = new ArrayList<>();
-                ResourceUtils.getBuildConfigFor(model.getResource(), buildConfigs);
-                projectsAndBuildConfigs.put(model.getResource().getProject(), buildConfigs);
+                if (model.getResource() != null) {
+                	projectsAndBuildConfigs.put(model.getResource().getProject(), Collections.emptyList());
+                }
```
Comment (adietish) [inline plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/server/ServerSettingsWizardPage.java]: this looks as if we wont store no build configs at all (if the model holds a resource, we store an empt list of buildconfigs for the project of that resource). We also dont get the build configs any more. Without testing it, this doesnt look correct. Is it?

Comment (bdshadow) [inline plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/server/ServerSettingsWizardPage.java]: the list of build configs is retrieved in `ImportApplicationWizard`
Look into lines 501-502 , before my changes - they do nothing. `ResourceUtils.getBuildConfigFor` was called and it is getter - so it returns some value. But why isn't it assigned to any variable and used after? This code is useless. Look inside `ResourceUtils.getBuildConfigFor` and examples of its usage in other places and you'll get the idea.
So my fix just removes the NPE indeed and nothing else. As you said, test it and you'll see that it works.

**Comment by bdshadow** [comment]
found that it causes some Exceptions later too. Need to fix it in  one more place too

**Comment by bdshadow** [comment]
@adietish can be tested now

**Comment by adietish** [comment]
@bdshadow other than the missing DCO things look fine to me. I'll merge it as soon as there's your required "signed-off by".

**Comment by bdshadow** [comment]
I've added "signed-off by" to the first commit, but forgot for the second one. Seems all of them need it.

**Comment by adietish** [comment]
@bdshadow yes absolutely. Given a PR that consists of several commits gets merged, I am not sure if the check is really helpful. Depends on the merge that would preserve the DCO or not (which I dont know)

**Comment by adietish** [comment]
manually merged both commits, made sure there's a signed-off and pushed to origin master in commit 0e5add768f1c7c5b7bdf1e7945fc96c118a4ecf0

**Pull request closed without merging**